### PR TITLE
feat(api): add endpoint to retrieve, update and delete events

### PR DESCRIPTION
**Pull request by arminpatel**

### CODE REVIEW

File: api/api/events/views.py [hunk 1]
```diff
@@ -29,3 +30,10 @@ def _has_create_permission(self, validated_data):
         user_roles = self.request.user.roles.filter(club__id=validated_data['club'].id)
 
         return (self.request.user.is_staff or (allowed_roles & user_roles).exists())
+
+
+class RetrieveUpdateDestroyEventsView(RetrieveUpdateDestroyAPIView):
```
Comment (Samy-33) [inline api/api/events/views.py:35]: Should `retrieving` be restricted to core members or admin?

**Comment by arminpatel** [comment]
Is there a way then to split this PR into smaller ones?

**Comment by Samy-33** [comment]
> Is there a way then to split this PR into smaller ones?

I am not sure if there's a pragmatic way to do that. But afaik, you can use git rebase to do that.

Or better, if the changes are per file basis: you could git rebase --soft and then choose which files to include in which commit.

**Comment by arminpatel** [comment]
I guess then we can't split this as retrieve, update and destroy have the same code only tests are different.

**Comment by Samy-33** [comment]
> Or better, if the changes are per file basis: you could git rebase --soft and then choose which files to include in which commit.

My bad I meant, `git reset --soft`

**Review by Samy-33 — DISMISSED** [review]

**Review by Samy-33 — APPROVED** [review]

**Pull request merged**